articles/build-an-outlook-clone-using-react-hooks/index.md

```diff
@@ -0,0 +1,881 @@
+
+1. **useState**: useState enables creating state variables and using it anywhere within the functional component. The syntax for useState is as shown below:
+
+   ```jsx
+    [stateName, setStateName] = useState(defaultValue)
+   ```
+    useState returns a list which consists of the state variable, and a function to set the state variable. Uses list destructuring in JavaScript, we can easily rename it to names according to our convenience.
+
+    The defaultValue can be any of the data types that JavaScript supports. We can have the stateName
+
+2. **useEffect**: useEffect is used for executing certain actions to take place upon triggering of events. useEffect replaces the original way of declaring lifecycle methods.
+   1. componentDidMount: To declare this event, we can use useEffect with an empty dependency list.
+   ```jsx
+   useEffect(()=>{
+       //update the state on componentDidMount
+   },[])
+   ```
+   2. componentWillUnMount: Once a component is unmounted, any cleanup that needs to be done can be performed by returning a function in useEffect.
+   ```jsx
+   useEffect(()=>{
+       //update the state on componentDidMount
+   return //cleanup code... for example freeing up event listeners
+   },[])
+   ```
+   3. dependency list: The dependency list ensures that the useEffect runs only when a change occurs in the state variables mentioned in the dependecy list. This way, we can modularize effects and direct it to different actions on the basis of state variables.
+   ```jsx
+   useEffect(()=>{
+       //update the state on componentDidMount
+   },[stateVariable1, stateVariable2])
+   ```
+   The above useEffect will run when there is a change in stateVariable1 or stateVariable2
+
+
+3. **useRef**: The useRef Hook is a function that returns a mutable ref object whose .current property is initialized with the passed argument (initialValue).
+   ```jsx
+    const refVariable = useRef(initialValue)
+   ```
+   It is usually used to access the DOM variables and to keep track of a mutable variable.
+4. **createContext** and **useContext**: createContext us used to pass props from parent to child components. Consider the example of passing props from parent to child components present many levels below. Sending it via the props makes the code messy. Instead, createContext is used to send the props. Whenever the props are needed, we use useContext to get the value of the props.  We will discuss this below in the project.
+
+### Code
+Alrighty, we first begin with creating a React project, using create-react-app package. Open a new terminal of your choice and execute the following commands.  
+
+```terminal
+npx create-react-app outlookclonereacthooks
+cd outlookclonereacthooks
+npm start
+```
```

Comment:
Never run a command when you don't know what it is intended to do. You must explain what you are doing with these commands to the reader.